v0/destinations/hs/util.js

```diff
@@ -376,6 +379,179 @@ const getEventAndPropertiesFromConfig = (message, destination, payload) => {
   return payload;
 };
 
+/**
+ *
+ * To reduce the number of calls for searching of already existing objects
+ * We do search for all the objects before router transform and assign the type (create/update)
+ * accordingly to context.hubspotOperation
+ *
+ * */
+
+const splitEventsForCreateUpdate = async (inputs, destination) => {
+  // get all the id and properties of already existing objects needed for update.
+  const updateHubspotIds = await getExistingData(inputs, destination);
+  const resultInput = [];
+
+  inputs.map(input => {
+    const { message } = input;
+    const { destinationExternalId } = getDestinationExternalIDInfoForRetl(
+      message,
+      "HS"
+    );
+
+    let filteredInfo = updateHubspotIds.filter(
+      update => update.property === destinationExternalId
+    );
+
+    if (filteredInfo.length) {
+      input.message.context.externalId = setHsSearchId(
+        input,
+        filteredInfo[0].id
+      );
+      input.message.context.hubspotOperation = "updateObject";
+      resultInput.push(input);
+    } else {
+      input.message.context.hubspotOperation = "createObject";
+      resultInput.push(input);
+    }
+  });
+
+  return resultInput;
+};
+
+const getHsSearchId = message => {
+  let externalIdArray = [];
+  let hsSearchId = null;
+  externalIdArray = message.context?.externalId;
+
+  if (externalIdArray) {
+    externalIdArray.forEach(extIdObj => {
+      const { type } = extIdObj;
+      if (type.includes("HS")) {
+        hsSearchId = extIdObj.hsSearchId;
+      }
+    });
+  }
+  return { hsSearchId };
+};
+
+const setHsSearchId = (input, id) => {
+  const { message } = input;
+  const resultExternalId = [];
+  if (message.context && message.context.externalId) {
+    externalIdArray = message.context.externalId;
```

Comment:
`externalIdArray` variable is not declared?

Reply:
How about `let externalIdArray = message.context?.externalId;` the way we are doing for getHsSearchId()